### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Evgenb000**

### CODE REVIEW

**Review by StasLukianchuk — APPROVED** [review]
GJ!

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filterPeople = century
+    ? people.filter(person =>
+      century === Math.ceil(person.died / 100) && (person.sex === 'm'))
+    : people.filter(person =>
+      person.sex === 'm');
```
Comment (SerhiiKirik) [inline src/averageAges.js:22]: you can make this check in filter callback
```suggestion
  const filterPeople = people.filter(person => century
    ? century === Math.ceil(person.died / 100) && (person.sex === 'm'))
    : person.sex === 'm');
```

File: src/averageAges.js [hunk 2]
```diff
@@ -37,7 +42,21 @@ function calculateMenAverageAge(people, century) {
...
+  const women = withChildren
+    ? people
+      .filter(person => person.mother !== null)
+      .map(person => person.mother)
+    : people.filter(person => person.sex === 'f');
+
+  const mothers = people.filter(person => women.includes(person.name));
+  const averageMothersAges = mothers.map(person => person.died - person.born);
+  const ages = women.map(person => person.died - person.born);
+  const averageWomenAges = withChildren
```
Comment (SerhiiKirik) [inline src/averageAges.js:54]: why did you check this two times?
You can just find women with children or all women, depending on the condition & calculate average age
Rewrite this

File: src/averageAges.js [hunk 3]
```diff
@@ -37,7 +42,21 @@ function calculateMenAverageAge(people, century) {
...
+    ? (averageMothersAges.reduce((acc, person) =>
+      acc + person, 0) / averageMothersAges.length)
+    : (ages.reduce((acc, person) => acc + person, 0) / ages.length);
+
+  return averageWomenAges;
```
Comment (SerhiiKirik) [inline src/averageAges.js:59]: Also, you can move average age calculation to the separate function

File: src/averageAges.js [hunk 4]
```diff
@@ -55,7 +74,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithSon = onlyWithSon
+    ? people
+      .filter(person => people.some(mother =>
+        mother.name === person.mother) && person.sex === 'm')
```
Comment (SerhiiKirik) [inline src/averageAges.js:80]: Move to the separate function
```suggestion
      .filter(person => hasMother(people, person.mother) && person.sex === 'm')
```

File: src/averageAges.js [hunk 5]
```diff
@@ -55,7 +74,22 @@ function calculateWomenAverageAge(people, withChildren) {
...
+    : people
+      .filter(person => people.some(mother =>
+        mother.name === person.mother));
```
Comment (SerhiiKirik) [inline src/averageAges.js:83]: Also, you can do like that
```suggestion
  const womenWithSon =  people.filter(person => onlyWithSon
    ? hasMother(people, person.mother) && person.sex === 'm')
    : hasMother(people,person.mother);
```

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Well done 🚀 
Let's improve your code 🧑‍💻

**Review by SerhiiKirik — APPROVED** [review]
Great 🔥

**Pull request closed without merging**